### PR TITLE
MGMT-6438 Remove references to unused env variables for hardware requiremets

### PR DESCRIPTION
**Pull request by jakub-dzon**
This PR removes last traces of env variables removed in #1715:
- HW_VALIDATOR_MIN_CPU_CORES_WORKER
- HW_VALIDATOR_MIN_CPU_CORES_MASTER
- HW_VALIDATOR_MIN_RAM_GIB_WORKER
- HW_VALIDATOR_MIN_RAM_GIB_MASTER
- HW_VALIDATOR_MIN_DISK_SIZE_GIB

Signed-off-by: Jakub Dzon <jdzon@redhat.com>

### CODE REVIEW

**Comment by jakub-dzon** [comment]
On hold until it's confirmed to be safe to merge from the cloud deployment perspective.

**Comment by jakub-dzon** [comment]
Got confirmation that removal from template.yaml should be done before cloud deployment is changed.

**Review by YuviGold — APPROVED** [review]
LGTM

**Pull request merged**
